### PR TITLE
design and basic integration of api's

### PR DESCRIPTION
**Pull request by srikanthgudipudi**

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -1,16 +1,28 @@
...
   },
+
   {
-    path:'',
-    redirectTo:'',
-    pathMatch:'full'
-  }
+    path: '',
+    // canActivate: [PrivateGuard],
+    loadChildren: () => import('./modules/full-layout/full-layout.module').then(m => m.FullLayoutModule)
```
Comment (kiranharidas187) [inline src/app/app-routing.module.ts:23]: It  should be renamed as a private module, as discussed

File: src/app/modules/community-core/interceptor/interceptor.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import { Injectable } from '@angular/core';
```
Comment (kiranharidas187) [inline src/app/modules/community-core/interceptor/interceptor.ts:1]: We planned to rename the module as portal-core and portal shared

File: src/app/modules/community-core/services/library-service/library.service.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  providedIn: 'root'
+})
+export class LibraryService {
+
+  constructor(private Http: HttpClient) { }
+
+
+  // To get the Filter List
+  getFilterList() {
+    return this.Http.get(environment.base_url + LibraryConfig.filterList);
```
Comment (kiranharidas187) [inline src/app/modules/community-core/services/library-service/library.service.ts:17]: Why are we not using the npm service which we created??

File: src/app/modules/community-shared/card/card.component.scss [hunk 4]
```diff
@@ -0,0 +1,29 @@
+.example-card {
```
Comment (kiranharidas187) [inline src/app/modules/community-shared/card/card.component.scss:1]: Why are you hardcoding the width?

File: src/app/modules/community-shared/card/card.component.scss [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  max-width: 400px;
+  min-width: 300px;
+}
+
+.example-header-image {
```
Comment (kiranharidas187) [inline src/app/modules/community-shared/card/card.component.scss:6]: give valid class names?

File: src/app/modules/community-shared/community-shared.module.ts [hunk 6]
```diff
@@ -4,15 +4,21 @@ import {MatButtonModule} from '@angular/material/button';
 import {MatIconModule} from '@angular/material/icon';
 import { CommunityFooterComponent } from './community-footer/community-footer.component';
 import { CommonModule } from '@angular/common';
+import { RouterModule } from '@angular/router';
```
Comment (kiranharidas187) [inline src/app/modules/community-shared/community-shared.module.ts:7]: Rename the module as portal-shared

File: src/app/modules/landing/landing-page/landing-page.component.html [hunk 7]
```diff
@@ -144,3 +144,5 @@ <h1 class="aboutText">Resources</h1>
 
   </mat-sidenav-content>
```
Comment (kiranharidas187) [inline src/app/modules/landing/landing-page/landing-page.component.html:145]: Why a this here?

File: src/app/modules/library/filters/filters.component.ts [hunk 8]
```diff
@@ -0,0 +1,272 @@
+import { Component, OnInit } from '@angular/core';
```
Comment (kiranharidas187) [inline src/app/modules/library/filters/filters.component.ts:1]: This component should be in shared.

File: src/app/modules/library/learning-resources/learning-resources.component.html [hunk 9]
```diff
@@ -0,0 +1,22 @@
+<div>
+    <mat-toolbar class="c-filters">
+        <div *ngFor="let data of filtersData">
+            <mat-form-field style="width: 250px;" class="example-chip-list">
```
Comment (kiranharidas187) [inline src/app/modules/library/learning-resources/learning-resources.component.html:4]: Why are you hardcoding the width?

File: src/app/modules/library/library-routing.module.ts [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+
+const routes: Routes = [{
+
+  path: '',
+  component: LibraryTabComponent,
+  children: [
+    { path: 'learning', component: LearningResourcesComponent },
+    { path: 'observation', component: ObservationsComponent },
+    { path: 'assesment', component: AssesmentsComponent },
```
Comment (kiranharidas187) [inline src/app/modules/library/library-routing.module.ts:16]: Where is the default path set?

File: src/app/modules/library/library-tab/library-tab.component.html [hunk 11]
```diff
@@ -0,0 +1,30 @@
+<div class="dashboard-content">
+  <p>
+    <mat-toolbar style="padding-left: 300px; background-color: white;">
```
Comment (kiranharidas187) [inline src/app/modules/library/library-tab/library-tab.component.html:3]: Multiple styles should go as a class

File: src/app/modules/library/library-tab/library-tab.component.html [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+      <div>
```
Comment (kiranharidas187) [inline src/app/modules/library/library-tab/library-tab.component.html:4]: Loop it using a static array

File: src/app/modules/library/library-tab/library-tab.component.ts [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+  templateUrl: './library-tab.component.html',
+  styleUrls: ['./library-tab.component.scss']
+})
+export class LibraryTabComponent implements OnInit {
+
+  tab: any = 'Learning';
+  constructor(private router: Router) { }
+
+  ngOnInit() {
+    this.router.navigate(['library/learning']);
```
Comment (kiranharidas187) [inline src/app/modules/library/library-tab/library-tab.component.ts:15]: This should not be done

File: src/app/modules/my-folder/sample/sample.component.html [hunk 14]
```diff
@@ -0,0 +1,5 @@
+<div class="dashboard-content">
```
Comment (kiranharidas187) [inline src/app/modules/my-folder/sample/sample.component.html:1]: Why us this component?

File: src/utils/app-init.ts [hunk 15]
```diff
@@ -0,0 +1,27 @@
+import { KeycloakService } from 'keycloak-angular';
```
Comment (kiranharidas187) [inline src/utils/app-init.ts:1]: Why is this

File: src/app/app-routing.module.ts (hunk 1 above)
Comment (srikanthgudipudi) [reply src/app/app-routing.module.ts:23]: Renamed

File: src/app/modules/community-core/interceptor/interceptor.ts (hunk 2 above)
Comment (srikanthgudipudi) [reply src/app/modules/community-core/interceptor/interceptor.ts:1]: renamed

File: src/app/modules/community-shared/card/card.component.scss (hunk 5 above)
Comment (srikanthgudipudi) [reply src/app/modules/community-shared/card/card.component.scss:6]: Removed

File: src/app/modules/community-shared/card/card.component.scss (hunk 4 above)
Comment (srikanthgudipudi) [reply src/app/modules/community-shared/card/card.component.scss:1]: The size of our width is different

File: src/app/modules/community-core/services/library-service/library.service.ts (hunk 3 above)
Comment (srikanthgudipudi) [reply src/app/modules/community-core/services/library-service/library.service.ts:17]: changed to npm usage

File: src/app/modules/community-shared/community-shared.module.ts (hunk 6 above)
Comment (srikanthgudipudi) [reply src/app/modules/community-shared/community-shared.module.ts:7]: Renamed

File: src/app/modules/landing/landing-page/landing-page.component.html (hunk 7 above)
Comment (srikanthgudipudi) [reply src/app/modules/landing/landing-page/landing-page.component.html:145]: That is for home page

File: src/app/modules/library/filters/filters.component.ts (hunk 8 above)
Comment (srikanthgudipudi) [reply src/app/modules/library/filters/filters.component.ts:1]: Moved to shared

File: src/app/modules/library/learning-resources/learning-resources.component.html (hunk 9 above)
Comment (srikanthgudipudi) [reply src/app/modules/library/learning-resources/learning-resources.component.html:4]: Default the  size of the mat-form-field is very small as per the design i changed

File: src/app/modules/library/library-tab/library-tab.component.html (hunk 11 above)
Comment (srikanthgudipudi) [reply src/app/modules/library/library-tab/library-tab.component.html:3]: changed

File: src/app/modules/library/library-tab/library-tab.component.html (hunk 12 above)
Comment (srikanthgudipudi) [reply src/app/modules/library/library-tab/library-tab.component.html:4]: Modified

File: src/app/modules/my-folder/sample/sample.component.html (hunk 14 above)
Comment (srikanthgudipudi) [reply src/app/modules/my-folder/sample/sample.component.html:1]: Removed

File: src/utils/app-init.ts (hunk 15 above)
Comment (srikanthgudipudi) [reply src/utils/app-init.ts:1]: we are tring for handling keycloak for private module right in that case i used

File: src/app/app-routing.module.ts [hunk 16]
```diff
@@ -1,16 +1,24 @@
 import { NgModule } from '@angular/core';
 import { Routes, RouterModule } from '@angular/router';
+import { PrivateGuard } from './modules/portal-core';
+import { initializer } from 'src/utils/app-init';
```
Comment (kiranharidas187) [inline src/app/app-routing.module.ts:4]: Use of this import

File: src/app/app.component.ts [hunk 17]
```diff
@@ -19,58 +21,64 @@ export class AppComponent {
   pushMode = 'side';
   currentUser;
   baseUrl;
-  logo =" ./assets/shikshalokam.png";
-  roleAcess=[];
-  constructor(private route : ActivatedRoute,private authService :AuthService ) {
+  logo = " ./assets/shikshalokam.png";
```
Comment (kiranharidas187) [inline src/app/app.component.ts:24]: Remove all unwanted changes

File: src/app/modules/landing/landing-page/landing-page.component.ts [hunk 18]
```diff
@@ -157,9 +159,10 @@ export class LandingPageComponent implements OnInit {
 
   ]
 
-  constructor() { }
+  constructor(private router: Router) { }
```
Comment (kiranharidas187) [inline src/app/modules/landing/landing-page/landing-page.component.ts:162]: Remove unwanted changes for this file

File: src/app/modules/library/learning-resources/learning-resources.component.scss [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+  background-color: white;
+}
+
+.c-card {
+  display: flex;
+  flex-wrap: wrap;
+  padding-top: 5%;
+}
+
+.example-header-image {
```
Comment (kiranharidas187) [inline src/app/modules/library/learning-resources/learning-resources.component.scss:13]: give valid name and what is the background image about?

File: src/app/modules/library/learning-resources/learning-resources.component.scss [hunk 20]
```diff
@@ -0,0 +1,50 @@
...
+  min-width: 300px;
+  flex-direction: column;
+  margin: 10px;
+}
+.icon-background {
+  background-color: #e5f3fe;
+  height: 70px;
+  width: 70px;
+  border-radius: 35px;
+  margin: 20% 35% 10% 35%;
```
Comment (kiranharidas187) [inline src/app/modules/library/learning-resources/learning-resources.component.scss:32]: Why is this big background here?

File: src/app/modules/library/learning-resources/learning-resources.component.scss [hunk 21]
```diff
@@ -0,0 +1,50 @@
...
+.c-image {
+  width: 30px;
+  margin-top: 25px;
+}
+.c-description {
+  text-align: center;
+}
+
+.c-view {
+  padding: 5% 0% 0% 50%;
```
Comment (kiranharidas187) [inline src/app/modules/library/learning-resources/learning-resources.component.scss:49]: Why is 50% here?

File: src/app/modules/my-folder/my-folder-routing.module.ts [hunk 22]
```diff
@@ -0,0 +1,16 @@
+import { NgModule, Component } from '@angular/core';
+import { Routes, RouterModule } from '@angular/router';
+import { SampleComponent } from './sample/sample.component';
+
+
+const routes: Routes = [{
```
Comment (kiranharidas187) [inline src/app/modules/my-folder/my-folder-routing.module.ts:6]: remove unwanted routes

File: src/app/modules/portal-core/services/library-service/library.service.ts [hunk 23]
```diff
@@ -0,0 +1,23 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+export class LibraryService {
+
+  constructor(private Http: HttpClient) { }
+
+
+  // To get the Filter List
+  getFilterList() {
```
Comment (kiranharidas187) [inline src/app/modules/portal-core/services/library-service/library.service.ts:16]: Are we using this service?

File: src/app/modules/portal-shared/card/card.component.scss [hunk 24]
```diff
@@ -0,0 +1,41 @@
...
+  width: 200px;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  height: 30px;
+}
+
+.mat-card img {
+  // object-fit: cover; /*this makes the image in src fit to the size specified below*/
+  // width: 100%; /* Here you can use wherever you want to specify the width and also the height of the <img>*/
+  height: 120%;
```
Comment (kiranharidas187) [inline src/app/modules/portal-shared/card/card.component.scss:14]: why is ite given  greater than 100%,  Please use VH to give height

File: src/app/modules/portal-shared/filters/filters.component.html [hunk 25]
```diff
@@ -0,0 +1,24 @@
...
+  <!-- <mat-toolbar class="c-filters"> -->
+  <div class="row" style="justify-content: space-evenly;padding-top: 50px;">
+    <!-- <form [formGroup]="filterForm"> -->
+    <!-- <mat-icon matTooltip="filters">assignment</mat-icon> -->
+    <div class="f-image">
+      <img src="/assets/images/filter.svg" alt="Photo of a Shiba Inu" class="svg">
+    </div>
+    <div class="tab-text" *ngFor="let data of filtersData">
+      <div class="col-md-4 col-lg-4">
+        <mat-form-field style="width:250px" class="example-chip-list">
```
Comment (kiranharidas187) [inline src/app/modules/portal-shared/filters/filters.component.html:11]: Should not hardcode width

File: src/app/modules/portal-shared/filters/filters.component.ts [hunk 26]
```diff
@@ -0,0 +1,48 @@
...
+      group.addControl(field.label, control);
+    });
+    return group;
+
+    // this.filterForm = this.fb.group({
+    //   filter: [],
+    // });
+  }
+
+  onchange() {
```
Comment (kiranharidas187) [inline src/app/modules/portal-shared/filters/filters.component.ts:43]: the function name should be came case

File: src/app/modules/private/private/private.component.html [hunk 27]
```diff
@@ -0,0 +1,3 @@
+<lib-page-layout [sideMenuData]="menudata" (sideMenuClick)="selectedMenu($event)"></lib-page-layout>
+
+<lib-headers [logindetails] ="userDetails" (logoutClick)="logoutMethod($event)"></lib-headers>
```
Comment (kiranharidas187) [inline src/app/modules/private/private/private.component.html:3]: headers should be inside lib-page-layout right?

File: src/assets/scss/responsive.scss [hunk 28]
```diff
@@ -1,258 +1,247 @@
...
-    }
-    ._halfWidth{
-        width: 100%;
-    }
-    .dashboard{
-        display: flex !important;
-        justify-content: center !important;
-        width:100%;
-    }
+@media only screen and (max-width: 600px) {
```
Comment (kiranharidas187) [inline src/assets/scss/responsive.scss:1]: is this responsive changes required ??

File: src/index.html [hunk 29]
```diff
@@ -1,17 +1,20 @@
...
+
 <head>
   <meta charset="utf-8">
   <title>Shikshalokam</title>
   <base href="/">
 
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <link rel="icon" type="image/x-icon" href="assets/shikshalokamfavicon.ico">
-  
+
```
Comment (kiranharidas187) [inline src/index.html:11]: remove unwanted chnages

File: src/styles.scss [hunk 30]
```diff
@@ -1,122 +1,230 @@
...
+  height: 20px;
+  width: 24px;
+  font-size: 30px;
+}
+
+.mat-drawer-side {
+  border: none !important;
+}
+
+.dashboard-content {
```
Comment (kiranharidas187) [inline src/styles.scss:134]: all classes should have _ in front for global custom class

File: src/styles.scss [hunk 31]
```diff
@@ -1,122 +1,230 @@
...
+  padding: 100px 50px 15px 50px;
```
Comment (kiranharidas187) [inline src/styles.scss:135]: Why is this much padding

File: src/styles.scss [hunk 32]
```diff
@@ -1,122 +1,230 @@
...
+//     margin-top: 0px;
+//     display: inline-block;
+//     text-align: center;
+//     margin-right: 0px;
+//     text-align: center;
+//     padding-left: 100px;
+//   }
+//   .c-sidenav {
+//     margin-top: 0px;
+//     margin-bottom: 0px;
```
Comment (kiranharidas187) [inline src/styles.scss:193]: remove unwanted coded

File: src/app/app-routing.module.ts (hunk 16 above)
Comment (srikanthgudipudi) [reply src/app/app-routing.module.ts:4]: Removed

File: src/app/app.component.ts (hunk 17 above)
Comment (srikanthgudipudi) [reply src/app/app.component.ts:24]: Removed

File: src/app/modules/landing/landing-page/landing-page.component.ts (hunk 18 above)
Comment (srikanthgudipudi) [reply src/app/modules/landing/landing-page/landing-page.component.ts:162]: Removed

File: src/app/modules/library/learning-resources/learning-resources.component.scss (hunk 19 above)
Comment (srikanthgudipudi) [reply src/app/modules/library/learning-resources/learning-resources.component.scss:13]: Not needed removed

File: src/app/modules/library/learning-resources/learning-resources.component.scss (hunk 20 above)
Comment (srikanthgudipudi) [reply src/app/modules/library/learning-resources/learning-resources.component.scss:32]: For Recently Added and Most Popular icons background color

File: src/app/modules/library/learning-resources/learning-resources.component.scss (hunk 21 above)
Comment (srikanthgudipudi) [reply src/app/modules/library/learning-resources/learning-resources.component.scss:49]: For view all button to place in middle of the cards

File: src/app/modules/library/library-routing.module.ts (hunk 10 above)
Comment (srikanthgudipudi) [reply src/app/modules/library/library-routing.module.ts:16]: default is the library component right ?

File: src/app/modules/my-folder/my-folder-routing.module.ts (hunk 22 above)
Comment (srikanthgudipudi) [reply src/app/modules/my-folder/my-folder-routing.module.ts:6]: Removed

File: src/app/modules/portal-core/services/library-service/library.service.ts (hunk 23 above)
Comment (srikanthgudipudi) [reply src/app/modules/portal-core/services/library-service/library.service.ts:16]: No, i am not using this service , we are using common npm service

File: src/app/modules/portal-shared/filters/filters.component.ts (hunk 26 above)
Comment (srikanthgudipudi) [reply src/app/modules/portal-shared/filters/filters.component.ts:43]: Changed

File: src/app/modules/private/private/private.component.html (hunk 27 above)
Comment (srikanthgudipudi) [reply src/app/modules/private/private/private.component.html:3]: yes,That is to pass data and to trigger logout

File: src/index.html (hunk 29 above)
Comment (srikanthgudipudi) [reply src/index.html:11]: Removed

File: src/styles.scss (hunk 31 above)
Comment (srikanthgudipudi) [reply src/styles.scss:135]: This is for to view content in the main panel, if we won't give much padding data is going under header

File: src/styles.scss (hunk 32 above)
Comment (srikanthgudipudi) [reply src/styles.scss:193]: Removed

File: src/app/modules/library/library-tab/library-tab.component.ts (hunk 13 above)
Comment (srikanthgudipudi) [reply src/app/modules/library/library-tab/library-tab.component.ts]: Removed

File: src/assets/scss/responsive.scss (hunk 28 above)
Comment (srikanthgudipudi) [reply src/assets/scss/responsive.scss:1]: Yes after bootstrap changes, still some changes not applied correctly

File: src/app/modules/portal-shared/filters/filters.component.html (hunk 25 above)
Comment (srikanthgudipudi) [reply src/app/modules/portal-shared/filters/filters.component.html]: here flex 1 is not working

File: src/styles.scss (hunk 30 above)
Comment (srikanthgudipudi) [reply src/styles.scss]: changed

File: src/app/modules/portal-shared/card/card.component.html [hunk 33]
```diff
@@ -0,0 +1,25 @@
+<div  #card (window:resize)="width=card.getBoundingClientRect().width">
```
Comment (kiranharidas187) [inline src/app/modules/portal-shared/card/card.component.html:1]: (window:resize) whay are we using this??

File: src/app/modules/portal-shared/filters/filters.component.html [hunk 34]
```diff
@@ -0,0 +1,24 @@
+<div style="padding-top:50px ;">
+  <form [formGroup]="filterForm">
+    <div class="row" formArrayName="filterSelects">
+
+      <div class="f-image">
+        <img style="cursor: pointer;" matTooltip="clear Filters" (click)="clearFilters()"
+          src="/assets/images/filter.svg" alt="Photo of a Shiba Inu" class="svg">
```
Comment (kiranharidas187) [inline src/app/modules/portal-shared/filters/filters.component.html:7]: give valid alt text

File: src/app/modules/portal-shared/filters/filters.component.scss [hunk 35]
```diff
@@ -0,0 +1,23 @@
...
+  color: $text-color;
+}
+
+.c-filters {
+  justify-content: center;
+  padding-top: 50px;
+}
+
+.field-width {
+  width: 280px;
```
Comment (kiranharidas187) [inline src/app/modules/portal-shared/filters/filters.component.scss:22]: Why is the width hardcoded

File: src/index.html [hunk 36]
```diff
@@ -1,27 +1,30 @@
...
   <link rel="manifest" href="manifest.json">
-  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.0/css/bootstrap.min.css">
+ <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.0/css/bootstrap.min.css">
   <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
   <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.0/js/bootstrap.min.js"></script>
 
-  <!-- <link rel="stylesheet" type="text/css" href="//cdn.jsdelivr.net/npm/slick-carousel@1.8.1/slick/slick.css"/>
-<link rel="stylesheet" type="text/css" href="//cdn.jsdelivr.net/npm/slick-carousel@1.8.1/slick/slick-theme.css"/>
+  <link rel="stylesheet" type="text/css" href="//cdn.jsdelivr.net/npm/slick-carousel@1.8.1/slick/slick.css" />
+  <link rel="stylesheet" type="text/css" href="//cdn.jsdelivr.net/npm/slick-carousel@1.8.1/slick/slick-theme.css" />
```
Comment (kiranharidas187) [inline src/index.html:18]: Why is this imports here?

File: src/index.html [hunk 37]
```diff
@@ -1,27 +1,30 @@
...
   <title>Shikshalokam</title>
   <base href="/">
 
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <link rel="icon" type="image/x-icon" href="assets/shikshalokamfavicon.ico">
-  
+
   <link rel="manifest" href="manifest.json">
-  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.0/css/bootstrap.min.css">
+ <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.0/css/bootstrap.min.css">
```
Comment (kiranharidas187) [inline src/index.html:13]: Please install it as an npm package and use

File: src/styles.scss [hunk 38]
```diff
@@ -1,122 +1,176 @@
...
+}
+.main-content {
+  padding: 20px;
+}
+
+mat-card {
+  word-break: break-word;
+}
+
+@media (max-width: 600px) {
```
Comment (kiranharidas187) [inline src/styles.scss:160]: Why is this media query?

File: src/app/modules/library/learning-resources/learning-resources.component.html [hunk 39]
```diff
@@ -0,0 +1,36 @@
...
+      </div>
+      <div class="col-md-9 col-lg-9 col-xs-12">
+        <span *ngFor="let rdata of data['resources']">
+          <div class="col-md-4 col-lg-4 col-xs-12">
+            <app-card [cardData]="rdata"></app-card>
+          </div>
+        </span>
+      </div>
+      <div class="c-viewmore">
+        <a href="{{data?.viewMoreUrl}}" target="_blank"><button style="width: 200px;" mat-button class="_p-button">View
```
[the remark this answers is not in the document]
Comment (srikanthgudipudi) [reply src/app/modules/library/learning-resources/learning-resources.component.html]: Changed

File: src/app/modules/portal-shared/card/card.component.html (hunk 33 above)
Comment (srikanthgudipudi) [reply src/app/modules/portal-shared/card/card.component.html:1]: Based on the size the text inside the card automatically take 3 dots

File: src/app/modules/portal-shared/filters/filters.component.scss (hunk 35 above)
Comment (srikanthgudipudi) [reply src/app/modules/portal-shared/filters/filters.component.scss]: Removed

File: src/app/modules/library/learning-resources/learning-resources.component.scss (hunk 21 above)
Comment (srikanthgudipudi) [reply src/app/modules/library/learning-resources/learning-resources.component.scss]: changed to bootstrap

File: src/app/modules/portal-shared/card/card.component.scss (hunk 24 above)
Comment (srikanthgudipudi) [reply src/app/modules/portal-shared/card/card.component.scss]: changed

File: src/app/modules/portal-shared/filters/filters.component.html (hunk 34 above)
Comment (srikanthgudipudi) [reply src/app/modules/portal-shared/filters/filters.component.html]: changed

File: src/index.html (hunk 36 above)
Comment (srikanthgudipudi) [reply src/index.html]: for home page

File: src/app/modules/library/learning-resources/learning-resources.component.html [hunk 40]
```diff
@@ -0,0 +1,36 @@
...
+    <div class="row">
+      <div class="col-md-2 col-sm-12 center">
+        <section>
+          <div class="icon-background">
+            <img src="/assets/images/{{data?.title}}.png" alt="Photo of a Shiba Inu" class="c-image">
+          </div>
+          <div>
+            <h4>{{data?.title}}</h4>
+            <div class="_text-color" style="margin-top: 20px;">
+              {{data?.description}}
```
Comment (kiranharidas187) [inline src/app/modules/library/learning-resources/learning-resources.component.html:15]: it should be title

File: src/app/modules/library/learning-resources/learning-resources.component.html [hunk 41]
```diff
@@ -0,0 +1,36 @@
...
+            </div>
+          </div>
+        </section>
+      </div>
+      <div class="col-md-1">
```
Comment (kiranharidas187) [inline src/app/modules/library/learning-resources/learning-resources.component.html:20]: why is this blank div?

File: src/app/modules/library/learning-resources/learning-resources.component.ts [hunk 42]
```diff
@@ -0,0 +1,52 @@
...
+export class LearningResourcesComponent implements OnInit {
+  filtersData: any;
+  resourceData: any;
+  spin: boolean = true;
+
+  constructor(private communityService: CommunityService) { }
+
+  ngOnInit() {
+    this.getFilters();
+    let data = ['', '', '', ''];
```
Comment (kiranharidas187) [inline src/app/modules/library/learning-resources/learning-resources.component.ts:22]: Why is this??

File: src/app/modules/library/library/library.component.html [hunk 43]
```diff
@@ -0,0 +1,20 @@
+<div class="row" style="justify-content: center;">
+
+  <div *ngFor="let data of tabData" class="desktop-tab">
+    <div class="col-md-12 col-lg-12 col-sm-12 col-xl-8">
+      <button [routerLink]="data.routerLink" (click)="commingSoon()" routerLinkActive="_p-button" mat-button
```
Comment (kiranharidas187) [inline src/app/modules/library/library/library.component.html:5]: Why is this commingSoon function here??

File: src/app/modules/portal-shared/community-footer/community-footer.component.scss [hunk 44]
```diff
@@ -44,8 +45,8 @@
 }
 
 .wrapper {
-    // max-width: 1200px;
-    // margin: 0 auto;
+    max-width: 1200px;
```
Comment (kiranharidas187) [inline src/app/modules/portal-shared/community-footer/community-footer.component.scss:48]: why is this hardcoding?

File: src/index.html [hunk 45]
```diff
@@ -1,27 +1,26 @@
...
-  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.0/css/bootstrap.min.css">
-  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
-  <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.0/js/bootstrap.min.js"></script>
-
-  <!-- <link rel="stylesheet" type="text/css" href="//cdn.jsdelivr.net/npm/slick-carousel@1.8.1/slick/slick.css"/>
-<link rel="stylesheet" type="text/css" href="//cdn.jsdelivr.net/npm/slick-carousel@1.8.1/slick/slick-theme.css"/>
-
-<script src="//code.jquery.com/jquery-3.3.1.min.js"></script>
-<script src="//unpkg.com/slick-carousel@1.8.1/slick/slick.js"></script> -->
+  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.5.0/css/bootstrap.min.css"
```
Comment (kiranharidas187) [inline src/index.html:12]: Please remove unwanted links

File: src/app/modules/library/learning-resources/learning-resources.component.html (hunk 40 above)
Comment (srikanthgudipudi) [reply src/app/modules/library/learning-resources/learning-resources.component.html:15]: already it is title only right ?

File: src/app/modules/library/learning-resources/learning-resources.component.html (hunk 41 above)
Comment (srikanthgudipudi) [reply src/app/modules/library/learning-resources/learning-resources.component.html:20]: For space i added here

File: src/app/modules/library/learning-resources/learning-resources.component.ts (hunk 42 above)
Comment (srikanthgudipudi) [reply src/app/modules/library/learning-resources/learning-resources.component.ts:22]: To pass the empty data in initial  call

File: src/app/modules/library/library/library.component.html (hunk 43 above)
Comment (srikanthgudipudi) [reply src/app/modules/library/library/library.component.html:5]: Removed and given in router link

File: src/app/modules/portal-shared/community-footer/community-footer.component.scss (hunk 44 above)
Comment (srikanthgudipudi) [reply src/app/modules/portal-shared/community-footer/community-footer.component.scss:48]: Removed

File: src/index.html (hunk 45 above)
Comment (srikanthgudipudi) [reply src/index.html:12]: Changes made

**Pull request merged**